dhis-2/dhis-support/dhis-support-hibernate/src/main/java/org/hisp/dhis/hibernate/HibernateGenericStore.java

```diff
@@ -182,10 +197,10 @@ protected final Query getQuery( String hql )
      * @param sql the sql query.
      * @return a SqlQuery instance.
      */
-    protected final SQLQuery getSqlQuery( String sql )
+    protected final NativeQuery getSqlQuery( String sql )
     {
-        SQLQuery query = getSession().createSQLQuery( sql );
-        query.setCacheable( cacheable );
+        NativeQuery query = getSession().createNativeQuery( sql );
+        query.setHint( HibernateUtils.HIBERNATE_CACHEABLE_HINT , true );
         return query;

```

Comment:
Shouldn't this be `setHint( HibernateUtils.HIBERNATE_CACHEABLE_HINT , cacheable );` ?